web_api/national_park/test_api.py

```diff
@@ -0,0 +1,44 @@
+import unittest
+from api import park_data_collection
+
+# In this test module, we import the api.py module and the Park class to test the park_data_collection function.
+from api import park_data_collection
+
+class TestApi(unittest.TestCase):
+
+# We test the function by calling it with different parameters.
+    def test_park_data_collection(self):
+        # Test with state = "ca", search_query = "", park_code = None, limit = 5
+
+        parks = park_data_collection("ca")
+        self.assertEqual(len(parks), 5)
+        self.assertEqual(parks[0].__class__.__name__, "Park")
+        self.assertIsNotNone(parks[0].full_name)
```

Comment:
Instead of assertIsNotNone, it would be better to assert the data has the value you expect it to.